pydantic/generics.py

```diff
@@ -43,7 +43,7 @@ def __class_getitem__(  # type: ignore
         model_name = concrete_name(cls, params)
         validators = gather_validators(cls)
         fields: Dict[str, Tuple[Type[Any], Any]] = {
-            k: (v, getattr(cls, k, ...)) for k, v in concrete_type_hints.items()
+            k: (v, cls.__fields__[k].default) for k, v in concrete_type_hints.items() if k in cls.__fields__
```

Comment:
At some point I changed `GenericModel` to inherit from `BaseModel`, but I forgot to account for all the effects of the `MetaModel` metaclass.

I wanted to use
```python
fields = {name: (field.type_, field.default) for name, field in cls.__fields__.items()}
```
but this results in getting the *value* type, e.g. for Dict (this broke `test_deep_generic`). If there is a better way to get the `field.type_`, let me know.